ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -54,6 +54,20 @@ + (BOOL)requiresMainQueueSetup
     return @[@"RNDeviceInfo_batteryLevelDidChange", @"RNDeviceInfo_batteryLevelIsLow", @"RNDeviceInfo_powerStateDidChange"];
 }
 
+- (NSDictionary *)constantsToExport {
+    return @{
+         @"uniqueId": [self getUniqueId],
+         @"deviceId": [self getDeviceId],
+         @"bundleId": [self getBuildId],
+         @"systemName": [self getSystemName],
+         @"systemVersion": [self getSystemVersion],
+         @"appVersion": [self getAppVersion],
+         @"buildNumber": [self getBuildNumber],
+         @"isTablet": [self isTablet],
+         @"appName": [self getAppName],
+     };
```

Comment:
We also need to make `hasNotch` sync.
Probably you can implement this logic on the native side or just to add 
```objectivec
@"brand": [self getBrand],
@"model": [self getModel],
```

And then just rewrite `hasNotch` on JS side
````diff
let notch: boolean;
-export async function hasNotch() {
+export function hasNotch() {
  if (!notch) {
-    let _brand = await getBrand();
-    let _model = await getModel();
    notch =
      devicesWithNotch.findIndex(
        item =>
          item.brand.toLowerCase() === _brand.toLowerCase() &&
          item.model.toLowerCase() === _model.toLowerCase()
      ) !== -1;
  }
  return notch;
}
```

Reply:
```diff
let model: string;
-export async function getModel() {
+export function getModel() {
  if (!model) {
    if (OS === 'ios') {
      let deviceName;
-      let _deviceId = RNDeviceInfo.getDeviceId();
+      let _deviceId = RNDeviceInfo.deviceId; 
      if (_deviceId) {
        deviceName = deviceNamesByCode[_deviceId];
        if (!deviceName) {
          // Not found on database. At least guess main device type from string contents:
          if (_deviceId.startsWith('iPod')) {
            deviceName = 'iPod Touch';
          } else if (_deviceId.startsWith('iPad')) {
            deviceName = 'iPad';
          } else if (_deviceId.startsWith('iPhone')) {
            deviceName = 'iPhone';
          } else if (_deviceId.startsWith('AppleTV')) {
            deviceName = 'Apple TV';
          }
        }
      }
      model = deviceName || 'unknown';
    } else if (OS === 'android' || OS === 'windows') {
-      model = await RNDeviceInfo.getModel();
+      model = RNDeviceInfo.model; // Add on the Android side
    } else {
      model = 'unknown';
    }
  }
  return model;
}
```

```diff
let brand: string;
-export async function getBrand() {
+export function getBrand() {
  if (!brand) {
    if (OS === 'android' || OS === 'windows') {
-      brand = await RNDeviceInfo.getBrand();
+      brand = RNDeviceInfo.brand; // Add on the Android side
    } else if (OS === 'ios') {
      brand = 'Apple';
    } else {
      brand = 'unknown';
    }
  }
  return brand;
}
```

And Finally
```diff
let notch: boolean;
-export async function hasNotch() {
+export function hasNotch() {
  if (!notch) {
-    let _brand = await getBrand();
-    let _model = await getModel();
+    let _brand = getBrand();
+    let _model = getModel();
    notch =
      devicesWithNotch.findIndex(
        item =>
          item.brand.toLowerCase() === _brand.toLowerCase() &&
          item.model.toLowerCase() === _model.toLowerCase()
      ) !== -1;
  }
  return notch;
}
```

Reply:
All these points are valid but I wasn't planning on moving things around to much before I got confirmation from @mikehardy that these properties are ok.

Removing the `async` declaration on functions should be paired with a pull request on android side or we will be leaving current `master` broken for android users.

Reply:
these properties are okay - sorry I've been on the 5hrs-avg-sleep work schedule recently so just haven't had time to do it myself. But I'm generally okay with all the properties people are mentioning